### PR TITLE
Document watershed APIs

### LINKED ISSUE #193: Document Melton Ratio
The watershed API docstring, which also appears in the autogenerated help, does not mention the Melton Ratio.

Possibly hydrology was going to provide some additional explanation / references for the Melton Ratio, which would go in the sphinx files, not the docstrings.

### LINKED ISSUE #216: document new APIs
The two new APIs, `watershed-streams` and `downstream`, aren't in the API document yet. Fix that.

### PR DESCRIPTION
**Pull request by corviday**
Updates documentation around the streamflow APIs:

* adds documentation for the `downstream` and `watershed_streams` APIs (resolve #216)
* adds documention of the Melton ration provided by @AreliaTW  to the `watershed` API (resolves #193)

@AreliaTW also provided images illustrating the Melton Ratio, but I am not sure how to include them. I shortened her text slightly for the purposes of an API document - API users don't need to know how the files are organized on the backend.

### CODE REVIEW

**Comment by corviday** [comment]
Here's the new documentation PDF. 

[climate-explorer-backend.pdf](https://github.com/pacificclimate/climate-explorer-backend/files/9903137/climate-explorer-backend.pdf)

**Comment by corviday** [comment]
Here's a zip file containing the html output of the new docs. The ones that have been updated are API docs for `watershed`, `downstream`, and `watershed_streams`.

[new_docs.zip](https://github.com/pacificclimate/climate-explorer-backend/files/9903157/new_docs.zip)

File: doc/source/api/watershed-api-usage.md [hunk 1]
```diff
@@ -18,4 +18,12 @@ However, for small areas with steep topography, it is common to see some of the
 
 Consider the most extreme case of `n = 2` cells that happen to be positioned at the edge of a steep sided valley. One cell is in the valley bottom with an average elevation of 100 m. The other is cell, just adjacent to it, mostly covers the highland above with an average elevation of 500 m. In a histogram with 100 m bin width, we'd see non-zero areas for the 100 m bin and the 500 m bin, but zero areas for the 200 m, 300 m, and 400 m elevation bins, and in the graph these would look like gaps.
```
Comment (rod-glover) [inline doc/source/api/watershed-api-usage.md:19]: > The other is cell

"The other cell" (delete "is).

File: doc/source/api/watershed-api-usage.md [hunk 2]
```diff
@@ -18,4 +18,12 @@ However, for small areas with steep topography, it is common to see some of the
...
 
-We can see a similar effect for other small values of `n > 2` in steep terrain too. Once `n` becomes large enough, then the likelihood of an elevation bin not having some cells is quite low and these gaps do not appear.
\ No newline at end of file
+We can see a similar effect for other small values of `n > 2` in steep terrain too. Once `n` becomes large enough, then the likelihood of an elevation bin not having some cells is quite low and these gaps do not appear.
+
+## Melton Ratio
+
+The Melton Ratio (ratio of watershed relief to the square of the area) is used to characterize a location as being dominated by either flooding (Melton ratios <0.3) or debris flows (Melton ratios >0.6) (Wilford et al., 2004).
```
Comment (rod-glover) [inline doc/source/api/watershed-api-usage.md:25]: Generally, I believe it is preferred to have a space both before and after an equality or inequality sign (e.g., prefer "Melton ratios < 0.3". There are a couple of instances of this.

File: doc/source/api/watershed-api-usage.md [hunk 3]
```diff
@@ -18,4 +18,12 @@ However, for small areas with steep topography, it is common to see some of the
...
+
+The watershed API delineates a watershed boundary, or drainage polygon, based on a pour point of interest in combination with VIC-GL’s flow network. The drainage polygon is used to select the minimum and maximum elevations, compute the watershed relief and area.
```
Comment (rod-glover) [inline doc/source/api/watershed-api-usage.md:27]: Do we need to define "pour point"? I _think_ I know what it means, but I'm not sure. Who is our real audience?

File: doc/source/api/watershed-streams-api-usage.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+This API endpoint provides a GeoJSON multiline object representing stream network connectivity upstream of the selected point. It is intended to allow the network of streams that drains to a specific outlet to be visualized on a map. No data about the size of the streams is provided, and streams are approximated as straight line segments connecting the centers of grid cells.
+
+The resolution of the network is limited by the resolution of the grid used to calculate the network. Each grid cell is defined as draining into one specific other grid cell. If there are multiple small streams in a grid cell, or small creeks or culverts smaller than a single grid cell, the stream network object returned by this API will not represent them accurately.
```
Comment (rod-glover) [inline doc/source/api/watershed-streams-api-usage.md:3]: Excellent and useful explanation

**Review by rod-glover — CHANGES_REQUESTED** [review]
The rendered HTML in the zip file doesn't include the MD files. Was this working before? Wonder if it would be best to switch over to all RST? Or maybe there's just a module or plugin missing?

**Comment by corviday** [comment]
Huh, the html looked fine on my workstation. I must have just left some necessary files out of the .zip.

File: doc/source/api/watershed-api-usage.md (hunk 3 above)
Comment (corviday) [reply doc/source/api/watershed-api-usage.md:27]: Hm, yeah. This section was adapted from documentation Arelia wrote as part of our final report to MOTI, and I reworked it lightly for API usage. I suspect in the context of the API, calling it the "selected point" makes more sense. Good catch!

**Review by rod-glover — APPROVED** [review]
LGTM

**Pull request merged**